### PR TITLE
Add eslint and prettier

### PR DESCRIPTION
**Pull request by marceloF5**
Hello people,
Thinking about organization and standardization I submit a simple configuration of `eslint` and `prettier`. And obviously, we can improve other things over time, but for the moment I saw reason to just go here. These configurations already support the project well the development experience and we are already unlocked to analyze indentation, for example, future reviews. So from that moment its just saves the file and everything is automatically adjusted. Also, I added `.idea` and `.vscode` in gitignore.

Screenshot 2020-11-05 at 10 56 09

### CODE REVIEW

**Comment by marceloF5** [comment]
I saw that recently was added `.editorconfig` in master. So I removed my changes regarding `.editorconfig`.

**Review by matheuskuster — APPROVED** [review]
What do you think about running an `eslint --fix` to already fix all the files in the project using this style guide?

**Comment by marceloF5** [comment]
> What do you think about running an `eslint --fix` to already fix all the files in the project using this style guide?

I don't believe in this running much. I have already added it to other projects and this script is not used very often by us developers in my opinion. I think it makes more sense to use `eslint --fix` together with `lint-staged`, that is, whenever there is a new commit, it does `eslint --fix`. I am preparing another PR on this and I can mention you when it stays ready

**Comment by matheuskuster** [comment]
@marceloF5 Please do.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    ],
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaFeatures": {
+            "jsx": true
+        },
+        "ecmaVersion": 11,
+        "sourceType": "module"
+    },
+    "plugins": ["react", "@typescript-eslint", "react-hooks"],
```
Comment (andrei9669) [inline .eslintrc.json:28]: shouldn't prettier also be in the plugins section?

**Comment by andrei9669** [comment]
I think it would be much better if instead of the .json file, it was .js file so that when needed, we could easily comment on/out rules.

File: .eslintrc.json (hunk 1 above)
Comment (marceloF5) [reply .eslintrc.json:28]: No, I shouldn't. It's only necessary in case there is some specific rule added in `rule`.

**Comment by marceloF5** [comment]
> Acho que seria muito melhor se, em vez do arquivo .json, fosse um arquivo .js, para que, quando necessário, pudéssemos facilmente comentar sobre as regras.

Is there another reason to do it? Although I don't think it's necessary, it's easily possible to comment in .json file.

**Comment by andrei9669** [comment]
> > Acho que seria muito melhor se, em vez do arquivo .json, fosse um arquivo .js, para que, quando necessário, pudéssemos facilmente comentar sobre as regras.
> 
> Is there another reason to do it? Although I don't think it's necessary, it's easily possible to comment in .json file.

sorry if it's off-topic but I would really like to see how you comment in .json files.

**Comment by einSelbst** [comment]
> sorry if it's off-topic but I would really like to see how you comment in .json files.
Sorry for chiming in, valid question but in this case there is https://eslint.org/docs/user-guide/configuring#comments-in-configuration-files

**Comment by andrei9669** [comment]
> > sorry if it's off-topic but I would really like to see how you comment in .json files.
> 
> Sorry for chiming in, valid question but in this case there is https://eslint.org/docs/user-guide/configuring#comments-in-configuration-files

huh, didn't know that. feels wrong, but okay

**Comment by githubjosh** [comment]
is this issue still being worked on? 100% agree this project would benefit from the addition of ES Lint. Running this project against my `.eslintrc` file I've found unused code, Rule of Hooks errors, ...

Here's my current file (pretty minimal):

```
// Code-quality rules: https://eslint.org/docs/rules/
// Formatting rules: https://prettier.io/docs/en/options.html
{
  "env": {
    "browser": true,
    "node": true
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2018,
    "sourceType": "module",
    "impliedStrict": true,
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "ignorePatterns": ["/node_modules/*", "/.next", "/out", "/build"],
  "plugins": ["react", "react-hooks", "jsx-a11y", "prettier"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:react/recommended",
    "plugin:jsx-a11y/recommended",
    "plugin:react-hooks/recommended",
    "plugin:prettier/recommended",
    "eslint-config-prettier",
    "prettier"
  ],
  "settings": {
    "import/extensions": [".js", ".jsx", ".ts", ".tsx"],
    "react": {
      "version": "detect"
    }
  },
  "rules": {
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "error",
    "dot-notation":  "error",
    "no-implicit-coercion":  "error",
    "prefer-object-spread":  "error",
    "react/prop-types": 0
  }
}
```

**Comment by lfades** [comment]
Closing in favor of https://github.com/vercel/commerce/pull/425

**Pull request closed without merging**